### ISSUE TITLE
AstroConda released, update installation instructions accordingly

### ISSUE THREAD

**Issue by mperrin** [spacetelescope/webbpsf#109]
<a href="https://github.com/josePhoenix"><img src="https://avatars3.githubusercontent.com/u/172886?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [josePhoenix](https://github.com/josePhoenix)**
_Wednesday Apr 27, 2016 at 17:41 GMT_
_Originally opened as https://github.com/mperrin/webbpsf/issues/109_

----

Per @pllim, we can now tell users about the `conda` packages @jhunkeler made

**Comment by mperrin** [comment]
<a href="https://github.com/josePhoenix"><img src="https://avatars3.githubusercontent.com/u/172886?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [josePhoenix](https://github.com/josePhoenix)**
_Wednesday Apr 27, 2016 at 17:41 GMT_

----

http://astroconda.readthedocs.org/en/latest/installation.html

**Comment by mperrin** [comment]
<a href="https://github.com/pllim"><img src="https://avatars2.githubusercontent.com/u/2090236?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [pllim](https://github.com/pllim)**
_Wednesday Apr 27, 2016 at 17:42 GMT_

----

Should wait for confirmation from @jhunkeler for this one. I don't think JWST stuff is moved over yet.

**Comment by mperrin** [comment]
<a href="https://github.com/mperrin"><img src="https://avatars0.githubusercontent.com/u/1151745?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mperrin](https://github.com/mperrin)**
_Wednesday Apr 27, 2016 at 17:48 GMT_

----

I also want to understand more about the handling of the data packages - I know Conda is user selectable for what gets installed, but is the _default_ behavior going to be for all users to get our 400 MB of reference files?  That may not make the most sense.

**Comment by mperrin** [comment]
<a href="https://github.com/pllim"><img src="https://avatars2.githubusercontent.com/u/2090236?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [pllim](https://github.com/pllim)**
_Wednesday Apr 27, 2016 at 17:49 GMT_

----

@jhunkeler -- Please clarify. Thanks.

**Comment by mperrin** [comment]
<a href="https://github.com/mperrin"><img src="https://avatars0.githubusercontent.com/u/1151745?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mperrin](https://github.com/mperrin)**
_Wednesday Apr 27, 2016 at 17:50 GMT_

----

@josePhoenix anybody who wants/requires stability should be using the REL versions not DEV. 

My understanding from reading http://astroconda.readthedocs.org/en/latest/faq.html is that AstroConda will function more or less like REL, and is therefore a good choice for most users?

**Comment by mperrin** [comment]
<a href="https://github.com/josePhoenix"><img src="https://avatars3.githubusercontent.com/u/172886?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [josePhoenix](https://github.com/josePhoenix)**
_Wednesday Apr 27, 2016 at 17:51 GMT_

----

Based on the [manifest](http://ssb.stsci.edu/astroconda/osx-64/) these packages are installed by default when following the recommended installation instructions for AstroConda:

```
astropy >=1.1
cfitsio ==3.370
d2to1 ==0.2.12
ds9 ==7.4
fftw ==3.3.4
htc_utils >=0.1
imexam ==0.5.2
numpy 1.10*
photutils ==0.2.1
poppy ==0.4.0
purge_path >=0.0.1
pyds9 ==1.8.1
pyfftw ==0.9.2
python 3.4*
stsci-data-analysis
stsci-hst
webbpsf ==0.4.0
webbpsf-data ==0.4.0
```

I would say let them install `webbpsf-data` until/unless we hear complaints about it. People installing Ureka generally wanted things to "just work", and I suspect these people will be using the `stsci` metapackage to bring in everything and the kitchen sink.

**Comment by mperrin** [comment]
<a href="https://github.com/josePhoenix"><img src="https://avatars3.githubusercontent.com/u/172886?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [josePhoenix](https://github.com/josePhoenix)**
_Wednesday Apr 27, 2016 at 17:52 GMT_

----

However, I just noticed it hasn't picked up 0.4.1 yet... another thing to ask @jhunkeler

**Comment by mperrin** [comment]
<a href="https://github.com/pllim"><img src="https://avatars2.githubusercontent.com/u/2090236?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [pllim](https://github.com/pllim)**
_Wednesday Apr 27, 2016 at 17:54 GMT_

----

Most STScI users who use WebbPSF _and_ JWST pipeline will naturally use DEV. So, it is best to make sure it works both in REL and DEV reasonably well.

**Comment by mperrin** [comment]
<a href="https://github.com/pllim"><img src="https://avatars2.githubusercontent.com/u/2090236?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [pllim](https://github.com/pllim)**
_Wednesday Apr 27, 2016 at 17:55 GMT_

----

Re: Data files -- It is a philosophy issue. There is no right answer.

**Comment by mperrin** [comment]
<a href="https://github.com/josePhoenix"><img src="https://avatars3.githubusercontent.com/u/172886?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [josePhoenix](https://github.com/josePhoenix)**
_Wednesday Apr 27, 2016 at 17:59 GMT_

----

I would suggest a `/grp/jwst/ote/webbpsf-data-dev` folder to be updated from our stash of source material when we make changes to the data. Then the `ssbdev` environment could include an `export` line pointing the `WEBBPSF_PATH` there. This would also improve cases like @mperrin needing my IFU PSF reference data to evaluate a pull request. The downside is we'll forget to update it, since it's far too slow to use over NFS in day-to-day dev work.

**Comment by mperrin** [comment]
<a href="https://github.com/mperrin"><img src="https://avatars0.githubusercontent.com/u/1151745?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mperrin](https://github.com/mperrin)**
_Wednesday Apr 27, 2016 at 17:59 GMT_

----

"So, it is best to make sure it works both in REL and DEV reasonably well."   

Agreed. We don't plan to leave it broken in DEV for any great length of time. This will probably just make us (me in particular?) get better about doing development not in master. 

Relatedly now that you added me to the space telescope github group, I was wondering if I should move the main webbpsf repository there instead of in my own account.

**Comment by mperrin** [comment]
<a href="https://github.com/josePhoenix"><img src="https://avatars3.githubusercontent.com/u/172886?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [josePhoenix](https://github.com/josePhoenix)**
_Wednesday Apr 27, 2016 at 18:00 GMT_

----

(+1 to putting it under SpaceTelescope. If you do, be sure to make a new empty repository and not 'fork' it from here.)

**Comment by mperrin** [comment]
<a href="https://github.com/pllim"><img src="https://avatars2.githubusercontent.com/u/2090236?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [pllim](https://github.com/pllim)**
_Wednesday Apr 27, 2016 at 18:03 GMT_

----

Do you use Travis CI religiously? If so, I think @sosey said that there are too many repo on Space Telescope or something, and that can mess with Travis CI jobs being queued up too long.

**Comment by mperrin** [comment]
<a href="https://github.com/mperrin"><img src="https://avatars0.githubusercontent.com/u/1151745?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mperrin](https://github.com/mperrin)**
_Wednesday Apr 27, 2016 at 18:04 GMT_

----

Hmm, yes. I have tests for webbpsf & poppy both running in my own Travis account.

**Comment by mperrin** [comment]
<a href="https://github.com/pllim"><img src="https://avatars2.githubusercontent.com/u/2090236?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [pllim](https://github.com/pllim)**
_Wednesday Apr 27, 2016 at 18:05 GMT_

----

In addition, when you move the repo, your forks from this one _might_ (I'm actually not 100% sure about this one) to disappear. Something to think about.

**Comment by mperrin** [comment]
<a href="https://github.com/josePhoenix"><img src="https://avatars3.githubusercontent.com/u/172886?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [josePhoenix](https://github.com/josePhoenix)**
_Wednesday Apr 27, 2016 at 18:05 GMT_

----

@jhunkeler how should we ensure new releases get packaged and distributed through AstroConda? Should we add "email jhunkeler" to the release checklist after we tag it?

**Comment by mperrin** [comment]
<a href="https://github.com/pllim"><img src="https://avatars2.githubusercontent.com/u/2090236?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [pllim](https://github.com/pllim)**
_Wednesday Apr 27, 2016 at 18:06 GMT_

----

@nmearl, did you have problem with forks when you moved `specviz` to Space Telescope?

**Comment by mperrin** [comment]
<a href="https://github.com/sosey"><img src="https://avatars3.githubusercontent.com/u/503615?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [sosey](https://github.com/sosey)**
_Wednesday Apr 27, 2016 at 18:10 GMT_

----

The problem with running CI on Travis is there's a limit to the number of tests an organization can be running at one time. We've hung travis in the past and had to kill off processes to get the tests to complete. My recommendation before was that people run tests on their private repos while they are developing to lessen the chance of overload. Also, remember to check occasionally that your tests aren't hung and blocking people, especially with the much larger number of repos we now have.

**Comment by mperrin** [comment]
<a href="https://github.com/jhunkeler"><img src="https://avatars0.githubusercontent.com/u/490981?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jhunkeler](https://github.com/jhunkeler)**
_Wednesday Apr 27, 2016 at 18:10 GMT_

----

@josePhoenix Fork https://github.com/astroconda/astroconda-contrib, update the webbpsf, webbpsf-data, and/or poppy recipes, then open a pull request. On my end it's just a simple merge and compilation.

EDIT: Wording

**Comment by mperrin** [comment]
<a href="https://github.com/mperrin"><img src="https://avatars0.githubusercontent.com/u/1151745?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mperrin](https://github.com/mperrin)**
_Wednesday Apr 27, 2016 at 18:16 GMT_

----

OK, let's hold off on any repo relocations for now. Too many potential complications with Travis and breaking existing fork links which I don't want to deal with, especially at the same time as the initial release of astroconda. Let's only change one big thing at a time. :-)

**Comment by mperrin** [comment]
<a href="https://github.com/pllim"><img src="https://avatars2.githubusercontent.com/u/2090236?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [pllim](https://github.com/pllim)**
_Wednesday Apr 27, 2016 at 18:17 GMT_

----

Re: Moving repo -- You can always open an issue about it and see if anyone complains :smile:

**Comment by mperrin** [comment]
<a href="https://github.com/josePhoenix"><img src="https://avatars3.githubusercontent.com/u/172886?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [josePhoenix](https://github.com/josePhoenix)**
_Wednesday Apr 27, 2016 at 18:19 GMT_

----

This would be how to do it, but I think concerns about limited Travis resources might be reason enough not to. https://help.github.com/articles/transferring-a-repository/#transferring-from-a-user-to-an-organization

**Comment by mperrin** [comment]
<a href="https://github.com/josePhoenix"><img src="https://avatars3.githubusercontent.com/u/172886?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [josePhoenix](https://github.com/josePhoenix)**
_Monday Aug 22, 2016 at 15:14 GMT_

----

Fixed in #126
